### PR TITLE
Thread PRNG key through wrap_jax_jit to prevent UnexpectedTracerError

### LINKED ISSUE #17: How do RNGs work in torchax
I’ve read the code, and it seems that we just let PyTorch manage the RNGs (I’ve tested it, and it works on eager mode), but I have a question.

I’m not entirely sure how this works, but in JAX we sometimes provide RNG keys (e.g., jax.random.key) as arguments to the loss function. Since these keys are JAX arrays, they’re tracer objects. When we generate random numbers (random boolean, for dropout mask)based on these keys, the results also become tracer objects, which are then captured in the computation graph. Shouldn’t there be an equivalent mechanism for this in the torchax?

### PR DESCRIPTION
**Pull request by ifding**
Random aten ops (dropout, bernoulli, randn, …) call env.get_and_rotate_prng_key() which mutates RuntimeProperty.prng to a DynamicJaxprTracer while jit-tracing. The tracer escapes the trace and contaminates env.prng_key; subsequent reads return a tracer instead of a usable key, and any later jax.jit invocation that re-traces and references the leaked tracer raises
jax.errors.UnexpectedTracerError.

This fixes the leak by threading the PRNG key through the jit boundary as the last positional input and trailing output. Random ops rotate on a scoped RuntimeProperty (popped on trace exit), the rotated key is returned as a jit output, and the env is refreshed outside jit so subsequent calls see the advanced key. The fix is transparent to callers — wrap_jax_jit's signature is unchanged.

Also adds a prng_key setter (the only mutation path was previously manual_seed) so wrap_jax_jit can refresh the env after each jit call.

Closes #17.

### CODE REVIEW

**Comment by ifding** [comment]
Hi torchax maintainers — this is my first contribution, so the CI checks need                                                                                    
  to be approved manually. Could someone with write access click "Approve and                                                                                      
  run workflows"? The PR closes #17 and adds 4 regression tests covering the                                                                                       
  fix.                                                                                                                                                             
                                                                                                                                                                   
  Thanks!

**Review by qihqi — APPROVED** [review]

**Comment by ifding** [comment]
@qihqi  thanks for the approval, I just fixed some test failures, please take a look again. This is my local test running log after fixes:

```
test/test_embedding.py::TestEmbeddingPaddingIdx::test_embedding_grad_with_padding_idx_jitted PASSED                                                    [  6%]
test/test_interop.py::InteropTest::test_j2t_autograd_backward PASSED                                                                                   [ 12%]
test/test_interop.py::InteropTest::test_j2t_autograd_forward PASSED                                                                                    [ 18%]
test/test_interop.py::InteropTest::test_mod_attr PASSED                                                                                                [ 25%]
test/test_interop.py::InteropTest::test_module_with_shared_weights PASSED                                                                              [ 31%]
test/test_interop.py::InteropTest::test_to_jax_device PASSED                                                                                           [ 37%]
test/test_interop.py::InteropTest::test_torch_jax_view_dtype PASSED                                                                                    [ 43%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_consecutive_calls_advance_prng PASSED                                                              [ 50%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_manual_seed_controls_output PASSED                                                                 [ 56%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_random_op_does_not_leak_tracer PASSED                                                              [ 62%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_setter_symmetric_with_getter PASSED                                                                [ 68%]
test/test_train.py::TrainTest::test_scan_module PASSED                                                                                                 [ 75%]
test/test_train.py::TrainTest::test_train_step_can_run PASSED                                                                                          [ 81%]
test/test_jittable_module.py::JittableModuleTest::test_functional_call_callable PASSED                                                                 [ 87%]
test/test_jittable_module.py::JittableModuleTest::test_isinstance_does_not_mix PASSED                                                                  [ 93%]
test/test_jittable_module.py::JittableModuleTest::test_isinstance_works PASSED                                                                         [100%]
```

**Review by ifding — COMMENTED** [review]
test/test_interop.py::InteropTest::test_j2t_autograd_backward PASSED                                                                                       [  3%]
test/test_interop.py::InteropTest::test_j2t_autograd_forward PASSED                                                                                        [  6%]
test/test_interop.py::InteropTest::test_mod_attr PASSED                                                                                                    [  9%]
test/test_interop.py::InteropTest::test_module_with_shared_weights PASSED                                                                                  [ 12%]
test/test_interop.py::InteropTest::test_to_jax_device PASSED                                                                                               [ 16%]
test/test_interop.py::InteropTest::test_torch_jax_view_dtype PASSED                                                                                        [ 19%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_consecutive_calls_advance_prng PASSED                                                                  [ 22%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_manual_seed_controls_output PASSED                                                                     [ 25%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_random_op_does_not_leak_tracer PASSED                                                                  [ 29%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_setter_symmetric_with_getter PASSED                                                                    [ 32%]
test/test_embedding.py::TestEmbeddingPaddingIdx::test_embedding_grad_with_padding_idx_jitted PASSED                                                        [ 35%]
test/test_train.py::TrainTest::test_scan_module PASSED                                                                                                     [ 38%]
test/test_train.py::TrainTest::test_train_step_can_run PASSED                                                                                              [ 41%]
test/test_jittable_module.py::JittableModuleTest::test_functional_call_callable PASSED                                                                     [ 45%]
test/test_jittable_module.py::JittableModuleTest::test_isinstance_does_not_mix PASSED                                                                      [ 48%]
test/test_jittable_module.py::JittableModuleTest::test_isinstance_works PASSED                                                                             [ 51%]
test/test_context.py::TestContext::test_buffer PASSED                                                                                                      [ 54%]
test/test_context.py::TestContext::test_different_manual_seed PASSED                                                                                       [ 58%]
test/test_context.py::TestContext::test_generator_seed PASSED                                                                                              [ 61%]
test/test_context.py::TestContext::test_jit_with_rng PASSED                                                                                                [ 64%]
test/test_context.py::TestContext::test_mode_context_manager PASSED                                                                                        [ 67%]
test/test_context.py::TestContext::test_mode_decorator PASSED                                                                                              [ 70%]
test/test_context.py::TestContext::test_override_property PASSED                                                                                           [ 74%]
test/test_context.py::TestContext::test_same_manual_seed PASSED                                                                                            [ 77%]
test/test_misc.py::MiscTest::test_extract_jax_kwargs PASSED                                                                                                [ 80%]
test/test_misc.py::MiscTest::test_random_with_tensor_input PASSED                                                                                          [ 83%]
test/test_misc.py::MiscTest::test_to_device PASSED                                                                                                         [ 87%]
test/test_misc.py::MiscTest::test_to_device_twice PASSED                                                                                                   [ 90%]
test_dist/test_mesh_util.py::MeshUtilTest::test_init_module_sharded PASSED                                                                                 [ 93%]
test_dist/test_mesh_util.py::MeshUtilTest::test_sharder_call PASSED                                                                                        [ 96%]
test_dist/test_mesh_util.py::MeshUtilTest::test_sharder_call_no_shardable PASSED                                                                           [100%]

**Comment by ifding** [comment]
cc @qihqi to help check again, the previous failure is related to ruff format, already fixed them. 
Thanks so much~

```
$ uvx ruff format --check torchax test test_dist
59 files already formatted
```

**Comment by qihqi** [comment]
hi @ifding can you rebase to the latest? for some reason the CI is stuck. sorry about that.

**Comment by ifding** [comment]
Hi @qihqi  sure, just rebased to the latest commit, and re-run all related tests again, please help take a look, thanks!


test/test_interop.py::InteropTest::test_j2t_autograd_backward PASSED                                                                              [  3%]
test/test_interop.py::InteropTest::test_j2t_autograd_forward PASSED                                                                               [  7%]
test/test_interop.py::InteropTest::test_mod_attr PASSED                                                                                           [ 11%]
test/test_interop.py::InteropTest::test_module_with_shared_weights PASSED                                                                         [ 15%]
test/test_interop.py::InteropTest::test_to_jax_device PASSED                                                                                      [ 19%]
test/test_interop.py::InteropTest::test_torch_jax_view_dtype PASSED                                                                               [ 23%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_consecutive_calls_advance_prng PASSED                                                         [ 26%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_manual_seed_controls_output PASSED                                                            [ 30%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_random_op_does_not_leak_tracer PASSED                                                         [ 34%]
test/test_interop.py::JaxJitPRNGThreadingTest::test_setter_symmetric_with_getter PASSED                                                           [ 38%]
test/test_embedding.py::TestEmbeddingPaddingIdx::test_embedding_grad_with_padding_idx_jitted PASSED                                               [ 42%]
test/test_context.py::TestContext::test_buffer PASSED                                                                                             [ 46%]
test/test_context.py::TestContext::test_different_manual_seed PASSED                                                                              [ 50%]
test/test_context.py::TestContext::test_generator_seed PASSED                                                                                     [ 53%]
test/test_context.py::TestContext::test_jit_with_rng PASSED                                                                                       [ 57%]
test/test_context.py::TestContext::test_mode_context_manager PASSED                                                                               [ 61%]
test/test_context.py::TestContext::test_mode_decorator PASSED                                                                                     [ 65%]
test/test_context.py::TestContext::test_override_property PASSED                                                                                  [ 69%]
test/test_context.py::TestContext::test_same_manual_seed PASSED                                                                                   [ 73%]
test/test_flax.py::FlaxTest::test_flax_functional_call PASSED                                                                                     [ 76%]
test/test_flax.py::FlaxTest::test_flax_module_nested PASSED                                                                                       [ 80%]
test/test_flax.py::FlaxTest::test_flax_simple PASSED                                                                                              [ 84%]
test/test_misc.py::MiscTest::test_extract_jax_kwargs PASSED                                                                                       [ 88%]
test/test_misc.py::MiscTest::test_random_with_tensor_input PASSED                                                                                 [ 92%]
test/test_misc.py::MiscTest::test_to_device PASSED                                                                                                [ 96%]
test/test_misc.py::MiscTest::test_to_device_twice PASSED                                                                                          [100%]
